### PR TITLE
PEOPLE-136 Fix skill model validation

### PR DESCRIPTION
**Pull request by jasiekkk**
# Ticket
https://netguru.atlassian.net/browse/PEOPLE-136

# Description
- previously it was allowed to bypass the **uniques** validation while updating a skill

### CODE REVIEW

File: spec/models/skill_spec.rb [hunk 1]
```diff
@@ -13,13 +14,37 @@
...
+    let(:other_category) { create(:skill_category) }
 
-    before { create(:skill, name: 'foo', skill_category: category) }
+    let!(:existing_skill) do
+      create(:skill, name: 'foo', skill_category: category)
+    end
 
-    it 'ensurs uniques by name & category' do
+    it 'ensurs uniques by name & category for new_record' do
       skill = build(:skill, name: 'foo', skill_category: category)
```
Comment (DariuszMusielak) [inline spec/models/skill_spec.rb]: `let`?

File: spec/models/skill_spec.rb [hunk 2]
```diff
@@ -13,13 +14,37 @@
...
       expect(skill.valid?).to be false
-      expect(skill.errors.messages).to eq :'name & skill_category' => ['must be uniq']
+      expect(skill.errors.messages)
+        .to eq :'name & skill_category' => ['must be uniq']
+    end
+
+    it 'ensurs uniques by name & category when skill is updated' do
```
Comment (DariuszMusielak) [inline spec/models/skill_spec.rb]: I see here two `context`s

File: spec/models/skill_spec.rb [hunk 3]
```diff
@@ -20,31 +20,37 @@
...
+      end
     end
 
-    it 'allows update of other attributes' do
-      existing_skill.update(rate_type: 'boolean', description: 'description')
-      expect(existing_skill.valid?).to be true
+    context 'when updated' do
+      let(:persisted_skill) { create(:skill, name: 'ROR', skill_category: category) }
+
+      it 'ensures uniques by name & category' do
```
Comment (DariuszMusielak) [inline spec/models/skill_spec.rb]: I was thinking about something like this: 
```
context 'with existing combination of name and category' do
  it 'is invalid' do
    persisted_skill.update(name: 'foo')
    expect(persisted_skill.valid?).to be false
    expect(persisted_skill.errors.messages)
      .to eq :'name & skill_category' => ['must be uniq']
  end

  it 'is invalid' do
    persisted_skill.update(skill_category: category)
    expect(persisted_skill.valid?).to be false
    expect(persisted_skill.errors.messages)
      .to eq :'name & skill_category' => ['must be uniq']
  end
end

context 'with new combination of name and category' do
  before { persisted_skill.update(name: 'foo', skill_category: other_category) }

  it 'is valid' do
    expect(persisted_skill.valid?).to be true
  end
end
```

File: spec/models/skill_spec.rb [hunk 4]
```diff
@@ -13,13 +14,60 @@
...
+        expect(skill.errors.messages)
+          .to eq :'name & skill_category' => ['must be uniq']
+      end
+    end
+
+    context 'when updated' do
+      let(:persisted_skill) { create(:skill, name: 'ROR', skill_category: category) }
+
+      context 'with existing combination of name and category' do
+
```
Comment (netguru-hound) [inline spec/models/skill_spec.rb]: Extra empty line detected at block body beginning.

**Pull request merged**